processing/src/main/java/io/druid/query/dimension/CascadeExtractionDimensionSpec.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.dimension;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.metamx.common.StringUtils;
+import io.druid.query.extraction.CascadeExtractionFn;
+import io.druid.query.extraction.ExtractionFn;
+import io.druid.segment.DimensionSelector;
+
+import java.nio.ByteBuffer;
+
+/**
+ */
+public class CascadeExtractionDimensionSpec implements DimensionSpec
```

Comment:
The only reason that I add new DimensionSpec is just to support array of ExtractionFn as an argument of ExtractionDimensionSpec.

If user feels comfortable to use CascadeExtractionFn directly, it would be OK just using ExtractionDimensionSpec with CascadeExtractionFn.
However, Json representation of CascadeExtractionFn is a little complicated,
so I thought new dimensionSpec that hides CascadeExtractionFn and accepts array of ExtractionFns is more convenient to users.
```
{
  "type": "cascade"
  "extractonFn": 
  {
    "type": "substr",
    "index": 0,
    "length": 3
  }
  "childExtractionFn" :
 {
  "type": "cascade"
  "extractonFn": 
    {
      "type": "concat",
      "prefix": "[",
      "postfix": "]"
    }
  }
}